ACPs/194-streaming-asynchronous-execution/README.md

```diff
@@ -0,0 +1,394 @@
+
+Gas limits are typically set higher than expected gas used to allow for a buffer should gas estimates be imprecise.
+A lower bound of 50% of the specified limit, for example, would allow for a 100% buffer on estimates without penalising the sender, while still disincentivising falsely high limits.
+
+#### Upper bound on queue DoS
+
+With $R$ (for rate) as already defined (gas capacity per second), we introduce $g_L$, $g_U$, and $g_C$ as gas _limit_, _used_ and _charged_ per transaction, respectively, and
+
+$$
+g_C := \max(g_U, \lambda \cdot g_L)
+$$
+
+where $\lambda$ enforces a lower bound on the charge.
+
+The actual gas excess $x_A$ has an upper bound of the worst-case excess $x_W$, both of which can be used  to calculate respective base fees $f_A$ and $f_W$ (the variable element of gas prices) from the existing exponential function:
+
+$$
+f := M \cdot \exp\left( \frac{x}{K} \right).
+$$
+
+Mallory is attempting to maximize the DoS ratio
+
+$$
+D := \frac{f_W}{f_A}
+$$
+
+by maximizing $\Sigma_{\forall i} (g_L - g_U)_i$ to maximize $x_W - x_A$.
+
+> [!TIP]
+> Although $D$ shadows a variable in ACP-176, that one is very different to anything here so there won't be confusion.
+
+Recall that the proportionality $p$ of capacity per second and target gas per second ($R = pT$) results in increasing excess such that
+
+$$
+x := x + \frac{g (p-1)}{p}.
+$$
+
+If we allow the queue to have a bandwidth of $R\lambda^{-1}$ gas/second and limit its length $\tau$ seconds or a back-pressure window of $\omega = \tau R \lambda^{-1}$ gas, we can derive an upper bound on the difference in the changes to worst-case and actual gas excess:
+
+$$
+\begin{align}
+\Delta x_A &\ge \lambda \cdot \omega \cdot \frac{p-1}{p} \\
+\Delta x_W &= \omega \cdot \frac{p-1}{p} \\
+\Delta x_W - \Delta x_A &\le (1-\lambda) \cdot \omega \cdot \frac{p-1}{p} \\
+&= (1-\lambda) \cdot \frac{\tau R}{\lambda} \cdot \frac{p-1}{p} \\
+&= \frac{1-\lambda}{\lambda} \cdot \tau \cdot p T \cdot \frac{p-1}{p} \\
+&= \frac{1-\lambda}{\lambda} \cdot \tau \cdot T \cdot (p-1).
+\end{align}
+$$
+
+Note that we can express Mallory's DoS quotient as:
+
+$$
+\begin{align}
+D &= \frac{f_W}{f_A} \\
+&= \frac{ M \cdot \exp \left( \frac{x_W}{K} \right)}{ M \cdot \exp \left( \frac{x_A}{K} \right)} \\
+& = \exp \left( \frac{x_W - x_A}{K} \right).
+\end{align}
+$$
+
+When the queue is empty (i.e. the execution stream has caught up with accepted transactions), the worst-case fee estimate $f_W$ is known to be the actual base fee $f_A$; i.e. $Q = \emptyset \implies D=1$. The previous bound on $\Delta x_W - \Delta x_A$ also bounds Mallory's ability such that:
+
+$$
+D \le \exp \left( \frac{\frac{1-\lambda}{\lambda} \cdot \tau \cdot T \cdot (p-1)}{K} \right).
```

Comment:
In ACP-176, it is defined that: $K = 87 * T$.

We should simplify: $\frac{T}{K} = \frac{T}{87 \cdot T} = \frac{1}{87}$